packages/saber-plugin-netlify-redirect/lib/index.js

```diff
@@ -13,50 +13,51 @@ exports.apply = api => {
     const outputDir = api.resolveCache('public')
     const publicDir = api.resolveCwd('public')
 
-    await new Promise(resolve => {
-      fs.readFile(publicDir + '/_redirects', (err, data) => {
-        let oldContent = ''
-        if (!err) {
-          oldContent = `${data.toString()} \n`
-        }
-
-        const getRoute = link => urlJoin(api.config.build.publicUrl, link)
-
-        const getPageContent = routes => {
-          let newContent = ''
-          routes.forEach(route => {
-            route = getRoute(route)
-            newContent += `${route.replace('.html', '')} ${route} \n`
-          })
-          return oldContent + newContent
-        }
-
-        const generateRedirects = async links => {
-          const _redirects = path.join(outputDir, '/_redirects')
-          log.info(`Generating ${path.relative(outputDir, _redirects)}`)
-          await fs.outputFile(_redirects, getPageContent(links), 'utf8')
-          resolve()
-        }
-
-        const routes = [...api.pages.values()].map(
-          page => page.attributes.permalink
-        )
-        const filteredRoutes = routes.filter(
-          x =>
-            !x.endsWith('/') &&
-            routes.every(a => {
-              if (a === x || a === '/') {
-                return true
-              }
-              if (a.length > x.length) {
-                return a.indexOf(x.replace('.html', '')) !== 0
-              }
-              return true
-            })
-        )
-
-        generateRedirects(filteredRoutes)
+    var oldContent = ''
+    try {
+      oldContent = await fs.readFile(publicDir + '/_redirects')
```

Comment:
I used  `--no-verify` because I keep getting
`
Possible race condition: oldContent might be reassigned based on an out dated value of oldContent.  require-atomic-updates
`

Reply:
@egoist is it necessary to solve this if so how?